scripts/bmm_examples/Example1_Zhang&Luck2008.R

```diff
@@ -69,74 +70,81 @@ head(data_ZL2008)
-  # Initializing the dependent variable
-  RespErr ~ 1,   
+ZL_formula <- bmf(
   # estimating fixed intercept & random intercept for kappa of the first von Mises
   kappa ~ 0 + setsize + (0 + setsize || subID), 
   # estimating fixed intercept & random intercept for the mixing proportion 
   # for the memory/target distribution (i.e., p_mem)
   thetat ~ 0 + setsize + (0 + setsize || subID)
 )
 
+# We can access the default priors generated for this model using the default_prior function
+default_prior(ZL_formula, data = data_ZL2008, model = ZL_model)
 
 ###############################################################################!
 # 3) Model estimation ----------------------------------------------------------
 ###############################################################################!
 
-# fit mixture model if there is not already a results file stored
-if (!file.exists(here("output/fit_E1_ZL2008.RData"))) {
-  # using the fit_model function, we pass the bmm formula for the mixture model
-  # prior specification and other constraints are taken care of within this function
-  # additionally any further arguments for brms can be passed as well
-  fit_ZL_mixModel <- fit_model(
-    formula = ZL_mixFormula_bmm, # specify formula for mixture model
-    data    = data_ZL2008,   # specify data used to estimate the mixture model
-    model_type = "2p", # select the two-parameter model for fitting
-    
-    # save all potentially relevant information
-    sample_prior = TRUE,
-    save_pars = save_pars(all = TRUE),
-    
-    # add brms settings
-    warmup = warmup_samples,
-    iter = warmup_samples + postwarmup_samples, 
-    chains = nChains,
-    
-    # control commands for the sampler
-    control = list(adapt_delta = adapt_delta, 
-                   max_treedepth = max_treedepth)
-  )
+# using the bmm function, we pass the bmm formula for the mixture model and
+# the specified bmmodel. All other constraints are taken care of within this function
+# additionally any further arguments for brms can be passed as well
+ZL_fit <- bmm(
+  formula = ZL_formula, # specify formula for mixture model
+  data    = data_ZL2008,   # specify data used to estimate the mixture model
+  model = ZL_model, # select the two-parameter model for fitting
   
-  # save results into file
-  save(fit_ZL_mixModel, 
-       file = here("output/fit_E1_ZL2008.RData"),
-       compress = "xz")
+  # save all potentially relevant information
+  sample_prior = TRUE,
+  save_pars = save_pars(all = TRUE),
   
-} else {
-  # load results file
-  load(file = here("output/fit_E1_ZL2008.RData"))
-}
+  # add brms settings
+  warmup = warmup_samples,
+  iter = warmup_samples + postwarmup_samples, 
+  chains = nChains,
+  cores = parallel::detectCores(),
+  
+  # control commands for the sampler
+  control = list(adapt_delta = adapt_delta, 
+                 max_treedepth = max_treedepth),
+  
+  backend = "cmdstanr",
+  refresh = 100,
+  file = here("output/fit_E1_ZL2008_bmm")
```

Comment:
when I run this I get an error "Error: Object loaded via 'file' is not of class 'bmmfit'."

I think you might have ran this with an older version of bmm before we implemented the file argument - before we did that, the file argument was passed to brms instead, and the saved object was a brmsfit instead of a bmmfit object. Can you rerun all scripts with the newest bmm version and check if the files saved are bmmfit objects?

Reply:
Weird, when I load the file on my machine then I do not get the error message. I checked and the version that I used to estimate the models was `v0.5.1` and for me the file is also loaded as a `bmmfit` object.

Reply:
Maybe the osf files are old? Or mine are? I'll try deleting all of them and re-download to see if my files are out dated and I'll let you know

Reply:
I downloaded the files from osf, and the problem persists. When I fit the model myself it saves the correct bmmfit object. I suspect you might have uploaded old files? If the fits on your local machine are bmmfits, can you reupload them to OSF?